CMakeLists.txt

```diff
@@ -107,10 +108,22 @@ if (NOT CMAKE_BUILD_TYPE)
   message ("No build type detected. Build type set to Release by default")
   set ( CMAKE_BUILD_TYPE Release )
 endif()
-
-set ( CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11 -fno-rtti -fvisibility-inlines-hidden" )
-set ( CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -DVC_DEBUG -fvisibility-inlines-hidden" )
-set ( CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -O3 -fPIC -fvisibility-inlines-hidden" )
+set ( CMAKE_CXX_STANDARD 14 )
+set ( CMAKE_CXX_STANDARD_REQUIRED ON )
+set ( CMAKE_CXX_EXTENSIONS OFF )
+set ( CMAKE_VISIBILITY_INLINES_HIDDEN ON )
+set ( CMAKE_POSITION_INDEPENDENT_CODE ON )
+
+# https://gcc.gnu.org/wiki/Visibility#line-14 - Much lower chance of symbol collision.
+set ( CMAKE_CXX_VISIBILITY_INLINES_HIDDEN ON )
+
+# CMAKE_INTERPROCEDURAL_OPTIMIZATION applies different optimizations depending on the compiler. It breaks cross compiler compatibility (e.g. g++ with clang)
+# Disabling link time optimization.
+# set ( CMAKE_INTERPROCEDURAL_OPTIMIZATION ON) 
+set ( CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fno-use-cxa-atexit -fno-rtti" )
+if (CMAKE_COMPILER_IS_GNUC OR CMAKE_COMPILER_IS_GNUCXX)
+ set ( CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -O3" )
+endif(CMAKE_COMPILER_IS_GNUC OR CMAKE_COMPILER_IS_GNUCXX)
```

Comment:
The code is quite self-explanatory, yet the reason behind this is not.
Can you add a line explaining it?

Also, if this is only for `-O3`, possibly we can set compiler properties instead of touching cmake flags

Reply:
My fault, -O3 is ok.

https://gitlab.kitware.com/cmake/cmake/-/issues/23374
There is still no cmake config as of today